### ISSUE TITLE
Warn for `port module` with no ports

### ISSUE THREAD

**Issue by Fresheyeball** [elm/compiler#1386]
``` elm
port module Foo exposing (..)

bar = 12
```

I feel the above module should be a compiler warning. Something like:

**No ports found in port module Foo
Why not make Foo a non-port module?**

**Comment by mgold** [comment]
It *is* a compiler warning:

```
-- SYNTAX PROBLEM --------------------------------------------------------------

It looks like the keyword `module` is being used as a variable.

2| port module Foo exposing (..)
              ^
Rename it to something else.
```

You're thinking of *effects* modules, which are for experts and have additional special syntax in the module declaration. No one is going to make that mistake.

**Comment by jvoigtlaender** [comment]
@mgold, you are completely wrong. Have you actually confirmed what you say by runing `elm-make` on the code @Fresheyeball gave?
```
> more .\port.elm
port module Foo exposing (..)

bar = 12

> elm-make .\port.elm
Success! Compiled 1 module.
Successfully generated index.html
```

**Comment by mgold** [comment]
Huh. I tried it on elm-lang/try. So now I have no idea what's going on.

**Comment by rgrempel** [comment]
I think elm-lang/try does some special things with respect to modules -- for instance, even:

```elm
module Foo exposing (..)

bar = 12
```

... is an error on elm-lang/try.

In effect, I think that elm-lang/try does something to set up whatever you enter as your main module, and doesn't allow for some other kind of module definition.

**Comment by jvoigtlaender** [comment]
What Ryan says. Deriving assertions as [above](https://github.com/elm-lang/elm-compiler/issues/1386#issuecomment-220054184) from testing on http://elm-lang.org/try only is a bad idea.

**Comment by jvoigtlaender** [comment]
Can we delete all this irrelevant discussion here, starting with @mgold's comment above, so that only the actual issue statement remains?

**Comment by Fresheyeball** [comment]
@mgold if its there already, I'm not seeing it, and I get the other compiler warnings.

**Comment by mgold** [comment]
I'm going to guess that a "port module" is magic even blacker than "effects modules" and Evan will tell us about them in his own time.

**Comment by Fresheyeball** [comment]
ANY usage of ports must be in a "port module", so it cant be that black. I think "port module" is just a module that has ports.

**Comment by jvoigtlaender** [comment]
Again wrong, Max. They are not black magic, and Evan says explicitly what they are, on http://guide.elm-lang.org/interop/javascript.html.

**Comment by mgold** [comment]
Alright, well, it would be helpful if they had gotten more than a passing reference in the announcement. I'd like to think I have my ear to the ground on this stuff, although I read more mailing lists/slack/GitHub than docs, and this is the first I've heard of it.

**Comment by jvoigtlaender** [comment]
Anyway, can you delete your comments above? They are not helping the purpose of this issue here.

**Comment by evancz** [comment]
The development version of the compiler recognizes this and gives an error now:

```
-- NO PORTS ----------------------------------------------------------- temp.elm

This module does not declare any ports, but it says it will:

1| port module Foo exposing (..)
   ^^^^^^^^^^^
Switch this to module and you should be all set!
```

So this should become available with 0.19.1 along with a bunch of other improvements / fixes.
